src/app/profile/payment-methods/payment-methods.component.js

```diff
@@ -0,0 +1,124 @@
+import angular from 'angular';
+import template from './payment-methods.tpl';
+import recurringGiftsComponent from './recurring-gifts/recurring-gifts.component';
+import profileService from 'common/services/api/profile.service.js';
+import paymentMethod from './payment-method/payment-method.component';
+import paymentMethodFormModal from 'common/components/paymentMethods/paymentMethodForm/paymentMethodForm.modal.component';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+import paymentMethodDisplay from 'common/components/paymentMethods/paymentMethodDisplay.component';
+
+class PaymentMethodsController {
+
+  /* @ngInject */
+  constructor($uibModal, profileService, $log, $timeout) {
+    this.log = $log;
+    this.$uibModal = $uibModal;
+    this.paymentMethod = 'bankAccount';
+    this.profileService = profileService;
+    this.loading = false;
+    this.submissionError = {error: ''};
+    this.successMessage = { show: false };
+    this.$timeout = $timeout;
+  }
+
+  $onDestroy(){
+    if(this.paymentMethodFormModal) {
+      this.paymentMethodFormModal.close();
+    }
+  }
+
+  $onInit(){
+    this.loading = true;
+    this.loadPaymentMethods();
+  }
+
+  loadPaymentMethods() {
+    this.profileService.getPaymentMethodsWithDonations()
+      .subscribe(
+        data => {
+          this.loading = false;
+          this.paymentMethods = data;
+        },
+        error => {
+          this.loading = false;
+          this.error = 'Failed retrieving payment methods';
+          this.log.error(this.error, error);
+        }
+      );
+  }
+
+  addPaymentMethod() {
+    this.paymentMethodFormModal = this.$uibModal.open({
+      component: 'paymentMethodFormModal',
+      backdrop: 'static',
+      size: 'new-payment-method-modal',
+      windowTemplateUrl: giveModalWindowTemplate.name,
+      resolve: {
+        onSubmit: () => this.onSubmit,
+        profileService: this.profileService,
+        submitted: false,
+        submissionError: this.submissionError,
+        log: this.log,
+        parentComponent: this
+      }
+    });
+    this.paymentMethodFormModal.result.then((data) => {
+      this.successMessage = {
+        show: true,
+        type: 'paymentMethodAdded'
+      };
+      this.paymentMethods.push(data);
+      this.$timeout(()=>{
+        this.successMessage.show = false;
+      },60000);
+    });
+  }
+
+  onSubmit(e) {
+    if(e.data) {
+      this.profileService.addPaymentMethod(e.data)
+        .subscribe((data) => {
+          this.submitted = false;
+          this.parentComponent.paymentMethodFormModal.close(data);
```

Comment:
@OzzieOrca , in this case PaymentMethodsController's function onSubmit has scope of PaymentMethodFormModalController which doesn't have access to the modal instance. But you have those bindings down at PaymentMethodFormModalController resolve and dismiss. I was trying to call 'this.dismiss' inside PaymentMethodsController's onSubmit function and still no luck. It is kind of tricky and I am out of options. Could you please take a look at it and see if you can figure out a better way to close the modal on successful save? Thanks, man!